src/checkers/inference/solver/GeneralSolver.java

```diff
@@ -103,8 +103,40 @@ public InferenceSolution solve(Map<String, String> configuration, Collection<Slo
                     useGraph, solveInParallel);
             PrintUtils.writeStatistic(StatisticRecorder.getStatistic(), modelRecord, backEndType,
                     useGraph, solveInParallel);
+            postStatistic(constraints, slots, inferResult);
         }
-        return solution;
+
+        postVerification(constraints, slots, inferResult, qualHierarchy);
```

Comment:
I think "verifySolution" would be better. Normally, postXXX is called in a method XXX. eg. postAsMemberOf() in asMemberOf().